src/download.rs

```diff
@@ -89,6 +84,21 @@ where
         }
     }
 
+    Ok(res)
+}
+
+pub async fn download_and_hash<U, W>(client: &reqwest::Client, url: U, mut data: W, print_progress: bool) -> Result<DownloadResult<W>>
```

Comment:
We can't split this function because at this point the download is not finished, it continues in `res.chunk().await` below.